src/s3/util/s3Tab.ts

```diff
@@ -5,92 +5,97 @@
 
 import * as vscode from 'vscode'
 //import { Bucket, DownloadFileRequest, File, S3Client } from '../../shared/clients/s3Client'
-
+import { S3BucketNode } from '../explorer/s3BucketNode'
+import { S3FileNode } from '../explorer/s3FileNode'
+import { S3FolderNode } from '../explorer/s3FolderNode'
+import { S3Client } from '../../shared/clients/s3Client'
+import { uploadWithProgress } from '../commands/uploadFile'
 //need to subscribe to ondidsave and upload to s3, might need the S3FileNode
 
+//const contentType = mime.lookup(path.basename(request.fileLocation.fsPath)) || DEFAULT_CONTENT_TYPE
+//FOUND IN : s3Client.ts
 export class S3Tab {
-    //private file: File
-    private s3Uri: vscode.Uri
+    public s3Uri: vscode.Uri
     private window: typeof vscode.window
     private editor: vscode.TextEditor | undefined
+    private parent: S3BucketNode | S3FolderNode
+    private s3Client: S3Client
 
-    //private editing: boolean
-
-    public constructor(private fileUri: vscode.Uri, window = vscode.window) {
+    public constructor(public fileUri: vscode.Uri, public s3FileNode: S3FileNode, window = vscode.window) {
         this.s3Uri = vscode.Uri.parse('s3:' + this.fileUri.fsPath)
+        this.parent = s3FileNode.parent
```

Comment:
Probably unnecessary since you can just use the node's parent directly (unless `s3FileNode` can get overwritten?)

Reply:
It doesn't get reassigned ever